src/Waypoint.cpp

```diff
@@ -5,7 +5,7 @@
 #include "Waypoint.h"
 
 Waypoint::Waypoint(const QStringList& stringList) {
-	if(stringList.size() != 3)
+    if(stringList.size() != 6)
```

Comment:
Let's make this more safe while we are at it. 
If this is not a valid input line, let's just die.
```suggestion
    if(stringList.size() != 6) {
        qCritical() << "Waypoint(): could not parse " << stringList << " as Waypoint. Expected 6 fields.";
        QTextStream(stdout) << "CRITICAL: Waypoint(): could not parse " << stringList << " as Waypoint. Expected 6 fields." << '\n';
        exit(EXIT_FAILURE);
    }
```
The `qCritical()` is there for having a proper log.txt entry.

Reply:
I dont think we should die. Not being able to parse one waypoint is not really affecting the operation of the application. Worst case the plotted route is not following the true route 100%.

Reply:
The data is bad then. It might lead to all kinds of hard to chase errors down the line. 
I strongly think someone should fix the data in that case. Giving that early feedback to the user is the best we can do IMHO.

Reply:
On afterthought I understand that fixing the navdata might be hard for users.
So let's just log it as qCritical.